unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/ECSRuntime/Components/ECSComponent.cs

```diff
@@ -12,12 +12,28 @@ public class ECSComponent<ModelType> : IECSComponent
             new DualKeyValueSet<IParcelScene, long, ECSComponentData<ModelType>>(50);
 
         private readonly Func<IECSComponentHandler<ModelType>> handlerBuilder;
-        private readonly Func<object, ModelType> deserializer;
+        internal readonly Func<object, ModelType> deserializer;
 
+        internal readonly IECSComponentPool<ModelType> iEcsComponentPool;
+
+        IEcsComponentImplementation implementation;
```

Comment:
I think design-wise it's strange that the component itself controls the pool capabilities. Normally, I would suggest that a pool controls component creation. 
Moreover, we have 2 additional abstractions `IECSComponentPool<ModelType>` and `IEcsComponentImplementation` which make things more complex than they should be. Also it looks like they are connected with each other but represented by different interfaces, it looks like a red flag.

Reply:
The problem here is that the `ECSComponent `class handles bot value types and reference types, and doing it in a generic way is tricky.
Because of the way SDK7's architecture handles `ECSComponents`, we are currently discerning between structs and classes because of the injected `ModelType `pools.
I know this is not the optimal way to design a clean code, but I couldn't find any other solution without a significant refactor, and that would take more time.